config/tf_modules/gcp-k8s-service/iam.tf

```
@@ -11,6 +11,13 @@ resource "google_service_account_iam_binding" "trust_k8s_workload_idu" {
   ]
 }
 
+resource "google_storage_bucket_iam_member" "bucket_get" {
+  count  = length(local.get_buckets)
+  bucket = local.get_buckets[count.index]
+  role   = "roles/storage.legacyBucketReader"
+  member = "serviceAccount:${google_service_account.k8s_service.email}"
```

```suggestion
  for_each = local.get_buckets
  bucket   = each.key
  role     = "roles/storage.legacyBucketReader"
  member   = "serviceAccount:${google_service_account.k8s_service.email}"
```

Use for_each so a change in the list of buckets doesn't drop permissions temporarily.